### PR TITLE
pueue 0.1.5 (new formula)

### PR DESCRIPTION
**Pull request by devster31**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
pueue 0.1.5 (new formula), homepage https://github.com/Nukesor/pueue, author @Nukesor

### CODE REVIEW

**Comment by pthariensflame** [comment]
Can you add a HEAD to the formula too?

File: Formula/pueue.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    system "cargo", "install", "--locked", "--root", prefix, "--path", "."
+    rm bin/"completions_pueue"
+
+    prefix.install_metafiles
+
+    bash_completion.install "utils/completions/pueue.bash" => "pueue.bash"
+    fish_completion.install "utils/completions/pueue.fish" => "pueue.fish"
+  end
+
+  plist_options :manual => "pueued"
```
Comment (pthariensflame) [inline Formula/pueue.rb]: I think this should be `pueued -d` for the purposes described here.

Comment (pthariensflame) [inline Formula/pueue.rb]:
```suggestion
  plist_options :manual => "pueued -d"
```

File: Formula/pueue.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+class Pueue < Formula
+  desc "Command-line tool for managing long-running shell commands"
+  homepage "https://github.com/Nukesor/pueue"
+  url "https://github.com/Nukesor/pueue/archive/v0.1.4.tar.gz"
+  sha256 "8e156d1037a88448219d9c5bdbc844799e616604985c86a83541f0a23481f302"
```
Comment (pthariensflame) [inline Formula/pueue.rb]:
```suggestion
  sha256 "8e156d1037a88448219d9c5bdbc844799e616604985c86a83541f0a23481f302"
  head "https://github.com/Nukesor/pueue.git"
```

**Review by pthariensflame — CHANGES_REQUESTED** [review]

File: Formula/pueue.rb (hunk 1 above)
Comment (devster31) [reply Formula/pueue.rb]: From what I understand the `-d` option forks the process. Using the `plist` as a supervisor this shouldn't be necessary.
The way I see this working is using `pueued -d` if I don't want to start the service with `brew services {run,start} pueue`, for example running only once.
I'd leave it running in the foreground and letting `launchd` do the supervision.

Comment (Nukesor) [reply Formula/pueue.rb]: Can confirm. Only use the `-d` flag, if you don't want `pueued` to be run by a service supervisor.

File: Formula/pueue.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  head "https://github.com/Nukesor/pueue.git"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", "--root", prefix, "--path", "."
+    rm bin/"completions_pueue"
```
Comment (Nukesor) [inline Formula/pueue.rb]: Just released v1.0.5 and removed this binary.
This has now been added to the normal build process with `build.rs` and will be executed when calling `cargo build`

**Review by Nukesor — CHANGES_REQUESTED** [review]

File: Formula/pueue.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  head "https://github.com/Nukesor/pueue.git"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", "--root", prefix, "--path", "."
+
+    prefix.install_metafiles
+
+    bash_completion.install "utils/completions/pueue.bash" => "pueue.bash"
```
Comment (pthariensflame) [inline Formula/pueue.rb]:
```suggestion
    bash_completion.install "utils/completions/pueue.bash" => "pueue"
```

File: Formula/pueue.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    fish_completion.install "utils/completions/pueue.fish" => "pueue.fish"
+  end
```
Comment (pthariensflame) [inline Formula/pueue.rb]:
```suggestion
    zsh_completion.install "utils/completions/_pueue" => "_pueue"
  end
```

**Review by pthariensflame — CHANGES_REQUESTED** [review]
Suggested some completion improvements.

File: Formula/pueue.rb (hunk 5 above)
Comment (devster31) [reply Formula/pueue.rb]: Not yet unfortunately, please see https://github.com/Nukesor/pueue/issues/57

Comment (pthariensflame) [reply Formula/pueue.rb]: Ah, I see.  :(

**Comment by chenrui333** [comment]
Thanks @devster31 for your contribution to Homebrew! :tada: 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**